ReactAndroid/src/main/jni/react/jni/OnLoad.cpp

```diff
@@ -646,30 +646,27 @@ static void loadScriptFromAssets(JNIEnv* env, jobject obj, jobject assetManager,
     "assetName", assetNameStr);
   #endif
 
-  env->CallStaticVoidMethod(markerClass, gLogMarkerMethod, env->NewStringUTF("loadScriptFromNetworkCached_read"));
+  env->CallStaticVoidMethod(markerClass, gLogMarkerMethod, env->NewStringUTF("loadScriptFromAssets_read"));
   executeApplicationScript(bridge, script, assetNameStr);
-  env->CallStaticVoidMethod(markerClass, gLogMarkerMethod, env->NewStringUTF("loadScriptFromNetworkCached_done"));
+  env->CallStaticVoidMethod(markerClass, gLogMarkerMethod, env->NewStringUTF("loadScriptFromAssets_done"));
 }
 
-static void loadScriptFromNetworkCached(JNIEnv* env, jobject obj, jstring sourceURL,
-                                   jstring tempFileName) {
+static void loadScriptFromFile(JNIEnv* env, jobject obj, jstring fileName, jstring sourceURL) {
   jclass markerClass = env->FindClass("com/facebook/react/bridge/ReactMarker");
 
   auto bridge = jni::extractRefPtr<Bridge>(env, obj);
-  std::string script = "";
-  env->CallStaticVoidMethod(markerClass, gLogMarkerMethod, env->NewStringUTF("loadScriptFromNetworkCached_start"));
-  if (tempFileName != NULL) {
-    script = react::loadScriptFromFile(jni::fromJString(env, tempFileName));
-  }
+  auto fileNameStr = fileName == NULL ? "" : fromJString(env, fileName);
+  env->CallStaticVoidMethod(markerClass, gLogMarkerMethod, env->NewStringUTF("loadScriptFromFile_start"));
+  auto script = fileName == NULL ? "" : react::loadScriptFromFile(fileNameStr);
+  auto sourceURLStr = sourceURL == NULL ? fileNameStr : fromJString(env, sourceURL);
   #ifdef WITH_FBSYSTRACE
-  auto sourceURLStr = fromJString(env, sourceURL);
   FbSystraceSection s(TRACE_TAG_REACT_CXX_BRIDGE, "reactbridge_jni_"
     "executeApplicationScript",
     "sourceURL", sourceURLStr);
   #endif
-  env->CallStaticVoidMethod(markerClass, gLogMarkerMethod, env->NewStringUTF("loadScriptFromNetworkCached_read"));
+  env->CallStaticVoidMethod(markerClass, gLogMarkerMethod, env->NewStringUTF("loadScriptFromFile_read"));
   executeApplicationScript(bridge, script, jni::fromJString(env, sourceURL));
```

Comment:
or actually, we should be using sourceURLStr instead of `jni::fromJString(env, sourceURL)`